graylog2-web-interface/src/views/components/searchbar/bookmark/BookmarkForm.test.jsx

```diff
@@ -0,0 +1,123 @@
+
+describe('BookmarkForm', () => {
+  describe('render the BookmarkForm', () => {
+    it('should render create new', () => {
+      const wrapper = renderer.create(<BookmarkForm value="new Title"
+                                                    onChangeTitle={() => {}}
+                                                    saveAsSearch={() => {}}
+                                                    disableCreateNew={false}
+                                                    toggleModal={() => {}}
+                                                    isCreateNew
+                                                    target={() => {}}
+                                                    saveSearch={() => {}} />);
+      expect(wrapper.toJSON()).toMatchSnapshot();
+    });
+
+    it('should render save', () => {
+      const wrapper = renderer.create(<BookmarkForm value="new Title"
+                                                    onChangeTitle={() => {}}
+                                                    saveAsSearch={() => {}}
+                                                    disableCreateNew={false}
+                                                    toggleModal={() => {}}
+                                                    isCreateNew={false}
+                                                    target={() => {}}
+                                                    saveSearch={() => {}} />);
+      expect(wrapper.toJSON()).toMatchSnapshot();
+    });
+
+    it('should render disabled create new', () => {
+      const wrapper = renderer.create(<BookmarkForm value="new Title"
+                                                    onChangeTitle={() => {}}
+                                                    saveAsSearch={() => {}}
+                                                    disableCreateNew
+                                                    toggleModal={() => {}}
+                                                    isCreateNew={false}
+                                                    target={() => {}}
+                                                    saveSearch={() => {}} />);
+      expect(wrapper.toJSON()).toMatchSnapshot();
+    });
+  });
+
+  describe('callbacks', () => {
+    it('should handle toggleModal', () => {
+      const onToggleModal = jest.fn();
+      const wrapper = mount(<BookmarkForm value="new Title"
+                                          onChangeTitle={() => {}}
+                                          saveAsSearch={() => {}}
+                                          disableCreateNew
+                                          toggleModal={onToggleModal}
+                                          isCreateNew={false}
+                                          target={() => {}}
+                                          saveSearch={() => {}} />);
+      wrapper.find('button[children="Cancel"]').simulate('click');
+      expect(onToggleModal.mock.calls.length).toBe(1);
```

Comment:
Instead of checking semi-internal structures that might change, we could also do:
```suggestion
      expect(onToggleModal).toHaveBeenCalled();
```

or

```suggestion
      expect(onToggleModal).toHaveBeenCalledTimes(1);
```